test/bound_natural_test.exs

```diff
@@ -0,0 +1,17 @@
+defmodule Orangeade.Generator.BoundNatural.Tests do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+  import Orangeade.Generator.BoundNatural, only: [stream: 1]
+
+  property "stream can't contain elements bigger than specified by limit" do
+    check all l <- integer(), l > 0 do
+      assert stream(limit: l) |> Caffeine.Stream.take(50) |> Enum.max() <= l
+    end
+  end
+
+  property "sum of generated elements is smaller or equal than multiplied limits" do
```

Comment:
This is good. I only came up with one property that loosely resembles your first one though I gave it almost no time. Streams might be a simple example but the practice will help with property-based testing for more complex things in the future.